tests/schedule_test.go

```diff
@@ -174,8 +193,9 @@ func (s *ScheduleFunctionalSuite) TestBasics() {
 
 	// create
 
+	ctx := s.newContext()
```

Comment:
This won't work... Go won't invoke the method of the subclass on a method of the base class.

You'll need to pass in a function pointer on instantiation or set a different dynamic configs per class on setup.